spring-core/src/main/java/org/springframework/util/comparator/BooleanComparator.java

```diff
@@ -63,8 +64,9 @@ public BooleanComparator(boolean trueLow) {
 
 
 	@Override
-	public int compare(Boolean v1, Boolean v2) {
-		return (v1 ^ v2) ? ((v1 ^ this.trueLow) ? 1 : -1) : 0;
```

Comment:
some time ago I had an answer about this here : https://stackoverflow.com/questions/55598277/is-there-a-useful-difference-between-p-q-and-p-q-for-booleans/55601399#55601399 and it turns out that the performance of XOR vs plain `!=` is same. This also makes it far easier to read